bin/build

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env ruby
+
+require "yaml"
+require "optparse"
+require "set"
+
+class Config
+  attr_reader :options
+
+  ##
+  # We need to preserve the order of answers, so we are using nested arrays
+  # rather than a hash. Note that the questions asked in later steps can depend
+  # on the answers to previous steps
+  #
+  VARIANTS = {
```

Comment:
@G-Rath @joshmcarthur I'm thinking we should pull this data structure out into a config file so that I can pull this script apart into single task scripts (which will be simpler and grow better. I'm thinking `build.yml` in the project root. Any objections?